conans/test/integration/extensions/test_deployers.py

```diff
@@ -0,0 +1,29 @@
+import textwrap
+
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+def test_deployer_custom_version_output():
+    deployer = textwrap.dedent("""
+    import os
+    def deploy(graph, output_folder, **kwargs):
+        # When running graph info with --requires, the root of the graph is a virtual node, skip it
+        if graph.root.recipe == "Cli":
+            ref_version = str(graph.nodes[1].ref)
```

Comment:
This seems a bit weird, and the access with ``[1]`` probably internal implementation detail that might break in the future.

Reply:
Yes, you're right. We need to think of a way to make this more handy